resource-manager/src/main/java/org/ihtsdo/otf/resourcemanager/ResourceManager.java

```diff
@@ -1,113 +1,353 @@
+	 * return {@code null} if the {@code Resource} does not exist.
+	 *
+	 * @param resourcePath Path to the resource.
+	 * @return an {@code InputStream} which corresponds to the resource stream.
+	 * Otherwise, it will return {@code null} if the {@code Resource} does not
+	 * exist.
+	 * @throws IOException If an error occurs while trying to load the resource.
+	 */
+	public InputStream readResourceStreamOrNullIfNotExists(final String resourcePath) throws IOException {
 		try {
-			String fullPath = getFullPath(resourcePath);
-			Resource resource = resourceLoader.getResource(fullPath);
+			final Resource resource = resourceLoader.getResource(getFullPath(resourcePath));
 			return resource.exists() ? resource.getInputStream() : null;
 		} catch (AmazonS3Exception e) {
 			throw new IOException("Failed to load resource '" + resourcePath + "'.", e);
 		}
 	}
 
-	public void writeResource(String resourcePath, InputStream resourceInputStream) throws IOException {
+	/**
+	 * Writes the data to the given resource path location.
+	 *
+	 * @param resourcePath        Which points to the resource that is
+	 *                            going to have the write operation performed
+	 *                            on it.
+	 * @param resourceInputStream Input stream which represents the converted
+	 *                            object, encoded as bytes.
+	 * @throws IOException If an error occurs while trying to write the
+	 *                     input stream to the specified resource path.
+	 */
+	public void writeResource(final String resourcePath,
+							  final InputStream resourceInputStream) throws IOException {
 		try {
-			try (OutputStream outputStream = writeResourceStream(resourcePath);
-				 InputStream inputStream = resourceInputStream) {
+			try (final OutputStream outputStream = openWritableResourceStream(resourcePath);
+				 final InputStream inputStream = resourceInputStream) {
 				StreamUtils.copy(inputStream, outputStream);
 			}
 		} catch (AmazonS3Exception e) {
 			throw new IOException("Failed to write resource '" + resourcePath + "'.", e);
 		}
 	}
 
-	public OutputStream writeResourceStream(String resourcePath) throws IOException {
+	/**
+	 * Retrieves the output stream, which will have the input stream written
+	 * into it.
+	 *
+	 * @param resourcePath Location of the resource.
+	 * @return {@code OutputStream} which is the content that can be written
+	 * to.
+	 * @throws IOException If an error occurs while trying to get the writable
+	 *                     resource {@code OutputStream}.
+	 */
+	public OutputStream openWritableResourceStream(final String resourcePath) throws IOException {
 		writeCheck();
-		String fullPath = getFullPath(resourcePath);
-		if (!resourceConfiguration.isUseCloud()) {
-			new java.io.File(fullPath).getParentFile().mkdirs();
+		final String fullPath = getFullPath(resourcePath);
+		if (!resourceConfiguration.isUseCloud()) new File(fullPath).getParentFile().mkdirs();
```

Comment:
Please include curly braces here. They are not strictly needed, it's just the code style we use for slightly better readability.

Reply:
Fixed